torch/csrc/profiler/api.h

```diff
@@ -139,6 +141,30 @@ struct TORCH_API CUDAStubs {
 TORCH_API void registerCUDAMethods(CUDAStubs* stubs);
 TORCH_API const CUDAStubs* cudaStubs();
 
+// ----------------------------------------------------------------------------
+// -- ITT ---------------------------------------------------------------------
+// ----------------------------------------------------------------------------
+using ITTEventStub = std::shared_ptr<CUevent_st>;
+
+struct TORCH_API ITTStubs {
+  virtual void record(int* device, ITTEventStub* event, int64_t* cpu_ns)
+      const = 0;
+  virtual float elapsed(const ITTEventStub* event, const ITTEventStub* event2)
+      const = 0;
+  virtual void ittMark(const char* name) const = 0;
+  virtual void ittRangePush(const char* name) const = 0;
+  virtual void ittRangePop() const = 0;
+  virtual bool enabled() const {
+    return false;
+  }
+  virtual void onEachDevice(std::function<void(int)> op) const = 0;
+  virtual void synchronize() const = 0;
+  virtual ~ITTStubs();
+};
```

Comment:
This looks like a word-by-word copy of `CUDAStubs` implemented just a few lines above, i.e.
https://github.com/pytorch/pytorch/blob/214951bc6bfff104483cde623289d04fa0b18588/torch/csrc/profiler/api.h#L121-L125

Why not rename `CUDAStubs` into something agnostic (say `ProfilerStubs`) and use them for both CUDA and ITT?

Reply:
Do you mean dispatch to CUDA or ITT inside each function?

Reply:
@malfet `CUDAStubs` is exposed in header files. Renaming it will cause changes to frontend API. Is it OK to do this?
./torch/include/torch/csrc/profiler/api.h:123:struct TORCH_API CUDAStubs {
./torch/include/torch/csrc/profiler/api.h:136:  virtual ~CUDAStubs();
./torch/include/torch/csrc/profiler/api.h:139:TORCH_API void registerCUDAMethods(CUDAStubs* stubs);
./torch/include/torch/csrc/profiler/api.h:140:TORCH_API const CUDAStubs* cudaStubs();